pyracf/common/security_request_error.py

```diff
@@ -18,9 +19,46 @@ def __init__(self, result: dict) -> None:
     def __str__(self) -> str:
         return self.message
 
+    def __check_result_type(self, result: dict) -> dict:
```

Comment:
Can we move this logic to `security_result.py`? All of that transformation of the result xml into a result dictionary should be handled in `security_result.py` with the exception of profile extract since those are all function group specific.

Reply:
That's a very fair point, I moved all of it over there as this is really about the security_result structure.